core/src/test/java/org/apache/iceberg/TestRewriteManifests.java

```diff
@@ -443,6 +444,14 @@ public void testBasicManifestReplacement() throws IOException {
     List<ManifestFile> manifests = snapshot.allManifests(table.io());
     Assert.assertEquals(3, manifests.size());
 
+    if (formatVersion == 1) {
+      assertThat(manifests.get(0).path()).isNotEqualTo(firstNewManifest.path());
+      assertThat(manifests.get(1).path()).isNotEqualTo(secondNewManifest.path());
+    } else {
+      assertThat(manifests.get(0).path()).isEqualTo(firstNewManifest.path());
+      assertThat(manifests.get(1).path()).isEqualTo(secondNewManifest.path());
+    }
```

Comment:
[minor] A comment might be helpful 
shouldn't we check if it's format = 1 and snapshotIdInheritence enabled ?

Reply:
There are actually two different tests. This one called basicReplacement and tests the default behavior. The one below explicitly validates enabled snapshot ID inheritance.